notes/lecture5.tex

```diff
@@ -0,0 +1,141 @@
+& \leq (1- \frac{2}{t+2}) (\frac{2\beta D^2}{t+2})+ \frac{4}{2(t+2)} \beta D^2 \\
+& = \beta D^2 (\frac{2t}{(t+2)^2} +\frac{2}{(t+2)^2}) \\
+& = 2\beta D^2 \frac{t+1}{(t+2)^2} \\
+& = 2\beta D^2 (\frac{t+1}{t+2}) (\frac{1}{t+2}) \\
+& \leq 2\beta D^2 (\frac{t+2}{t+3}) (\frac{1}{t+2}) \\
+& = 2\beta D^2 \frac{1}{t+3}
+\end{align*}
+Thus, the inequality also holds for the t+1 case.
+
+\end{proof}
+
+\subsection{Examples}
+The code for the following examples can be found \href{https://ee227c.github.io/code/lecture4.html}{here}.
+
+\subsubsection{Nuclear norm projection}
+The \textit{nuclear norm} (sometimes called \textit{Schatten $1$-norm} or \textit{trace norm}) of a matrix $A$, denoted $\|A\|_*$, is defined as the sum of its singular values
+\[
+\|A\|_* = \sum_i \sigma_i(A)\,.
+\]
+The norm can be computed from the singular value decomposition of $A$.
+We denote the unit ball of the nuclear norm by 
+\[
+B_*^{m\times n}=\{A\in\mathbb{R}^{m\times n} \mid \|A\|_*\le 1\}.
+\]
+How can we project a matrix $A$ onto $B_*$? Formally, we want to solve
+\[
+\min_{X\in B_*}\|A-X\|_F^2
+\]
+Due to the rotational invariance of the Frobenius norm, the solution is obtained by projecting the singular values onto the unit simplex. This operation corresponds to shifting all singular values by the same parameter $\theta$ and clipping values at $0$ so that the sum of the shifted and clipped values is equal to $1$. This algorithm can be found from \cite{Duchi2008}.
+
+\subsubsection{Low-rank matrix completion}
+Suppose we have a partially observable matrix $Y$ and we would like to 
+find its completion form projected in a nuclear norm ball. Formally we have the objective function
+\[
+\min_{X\in B_*}\frac{1}{2}\|Y-X \odot O\|_F^2
+\]
+And calculate the gradient of this function we will have
+\[
+\nabla f(X) = Y-X \odot O
+\]
+We can use projected gradient descent to solve this problem but it is more efficient to use Frank-Wolfe algorithm. We need to solve the linear optimization oracle
+\[
+\bar{X}_t\in \argmin_{X\in{B_*}}f(X_t)+\nabla f(X_t)^\trans(X-X_t)
+\]
+This will lead to a rank-1 matrix which is decomposed from $-\nabla f(X_t)$. This can be derived from \lemmaref{lemma}. Now we can have the update rule for the conditional gradient as
+\[
+X_{t+1} = X_t + \eta_t(-u_1v_1^\trans - X_t)
+\]
+where $u_1$ and $v_1$ are the top left and right singular vectors.\\ \\
+
+\begin{lemma}
+\lemmalabel{lemma}
+\[
+\min_{X\in B_*}\langle\nabla f(X_t)^\trans,X\rangle,\quad B_*=\{X|\|X\|_*\leqslant 1\}
+\]
+The optimal result is \[-\|\nabla f(X_t)\|\] with $X=-u_1v_1^\trans$ being one of the X that minimize the function where
+$u_1$ and $v_1$ are left and right singular vectors corresponding to the max singular value, $\|\nabla f(X_t)\|=\max_{\|X\|_*\leqslant 1}<\nabla f(X_t),X>$.\\
+\end{lemma}
+
+\begin{proof} [Proof of \lemmaref{lemma}.]
+Since the dual norm of a nuclear norm is operator norm,
```

Comment:
This is not the proof I gave in the lecture and we haven't introduced dual norms. The proof I gave was based on the fact that the nuclear norm ball is the convex hull of rank-1 matrices. The minimium is attained at a vertex, corresponding to a rank-1 matrix. From there a one line transformation shows that the problem is solved by the top left and right singular vectors of the matrix.

Reply:
Our original version was based on the nuclear norm ball is the convex hull of rank-1 matrices. But we do not have a solid proof of this fact. Do you have any suggestions on references about this proof?

Reply:
It follows from the singular value decomposition. You can state it as a fact without proof.